src/Generator/Generators/CSharp/CSharpTextTemplate.cs

```diff
@@ -962,6 +984,19 @@ private void GeneratePropertyGetter<T>(QualifiedType returnType, T decl, Class @
                     ReturnType = decl.QualifiedType
                 };
 
+                var arrayType = field.Type as ArrayType;
+
+                if (arrayType != null && @class.IsValueType)
+                {
+                    CSharpTypePrinter typePrinter = new CSharpTypePrinter(ctx.Driver);
+                    string type = arrayType.CSharpType(typePrinter).Type;
+                    WriteLine(string.Format("fixed ({0} __arrPtr = {1}.{2})",
```

Comment:
Type printers can have their contexts changed. You should be able to give it a native context and have the type produced right away instead of replacing [] with *.